### PR TITLE
Improve read_file for Windows

### PR DESCRIPTION
**Pull request by peterjc123**
Fixes https://github.com/pytorch/vision/issues/2743.

### CODE REVIEW

File: torchvision/csrc/cpu/image/read_write_file_cpu.cpp [hunk 1]
```diff
@@ -13,8 +20,17 @@ torch::Tensor read_file(std::string filename) {
 
   TORCH_CHECK(size > 0, "Expected a non empty file");
 
+#ifdef _WIN32
+  torch::Tensor data;
+  {
+    auto data_orig =
+        torch::from_file(filename, /*shared=*/false, /*size=*/size, torch::kU8);
+    data = std::move(data_orig.detach().clone());
```
Comment (fmassa) [inline torchvision/csrc/cpu/image/read_write_file_cpu.cpp:28]: Given that we are cloning the tensor, do we still need the move?
And is the `detach()` necessary given that we have created the tensor right away, so there is no history?

File: torchvision/csrc/cpu/image/read_write_file_cpu.cpp [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+#define VISION_STAT stat
+#endif
+
 torch::Tensor read_file(std::string filename) {
-  // CHECK if this only works on Windows for files smaller than 2GB
-  // https://stackoverflow.com/questions/5840148/how-can-i-get-a-files-size-in-c
-  struct stat stat_buf;
-  int rc = stat(filename.c_str(), &stat_buf);
+  struct VISION_STAT stat_buf;
+  int rc = VISION_STAT(filename.c_str(), &stat_buf);
```
Comment (fmassa) [inline torchvision/csrc/cpu/image/read_write_file_cpu.cpp:14]: Thanks for fixing this!

**Review by fmassa — COMMENTED** [review]
Thanks for the quick fix @peterjc123 !

I have one question, let me know what you think

File: torchvision/csrc/cpu/image/read_write_file_cpu.cpp [hunk 3]
```diff
@@ -1,10 +1,17 @@
 #include "read_write_file_cpu.h"
 
+// According to
+// https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/stat-functions?view=vs-2019,
+// we should use _stat64 for 64-bit file size on Windows.
+#ifdef _WIN32
+#define VISION_STAT _stat64
+#else
+#define VISION_STAT stat
```
Comment (fmassa) [inline torchvision/csrc/cpu/image/read_write_file_cpu.cpp:9]: nit: I wonder if best practices for macros is to  `#undef` those after they are used

File: torchvision/csrc/cpu/image/read_write_file_cpu.cpp (hunk 1 above)
Comment (peterjc123) [reply torchvision/csrc/cpu/image/read_write_file_cpu.cpp:28]: @fmassa This `move` is used to avoid calling the copy constructor/assignment in line 28. Do you mean we can actually do `data = torch::from_file(filename, /*shared=*/false, /*size=*/size, torch::kU8).clone()`?

Comment (fmassa) [reply torchvision/csrc/cpu/image/read_write_file_cpu.cpp:28]: Yes, that was my thinking. Is there any benefit from doing as it is currently done?

Comment (peterjc123) [reply torchvision/csrc/cpu/image/read_write_file_cpu.cpp:28]: @fmassa Nope. Will change according to your suggestion.

File: torchvision/csrc/cpu/image/read_write_file_cpu.cpp (hunk 3 above)
Comment (peterjc123) [reply torchvision/csrc/cpu/image/read_write_file_cpu.cpp:9]: Strictly speaking, yes. But it is okay since it is in a source file.

**Review by fmassa — APPROVED** [review]
Thanks a lot!
